packages/runtime/src/runtime/entry.tsx

```diff
@@ -3,13 +3,19 @@ import { Outlet, useNavigate, useLocation } from "react-router-dom";
 import { useEffect } from "react";
 import { SideBarMenu } from "./menu";
 import { EnsembleScreen } from "./screen";
+import { browserHistory } from "./history";
 
 interface EnsembleEntryProps {
   entry: EnsembleEntryPoint;
 }
 export const EnsembleEntry: React.FC<EnsembleEntryProps> = ({ entry }) => {
   const location = useLocation();
   const navigate = useNavigate();
+
+  // store copy of navigate and location for the use of outside components
+  browserHistory.navigate = navigate;
```

Comment:
The react router navigate function alone is not sufficient. We also need ability to pass inputs and validate the screen etc. You need to replicate/refactor the useNavigateScreen hook

Reply:
Done